daos-stack/daos
# DAOS-12287 test: CR Pass 4 - Orphan container

Test steps:
1. Create a pool and a container.
2. Inject fault to cause orphan container. i.e., container is left in the system, but doesn't appear with daos commands.
3. Check that the container doesn't appear with daos command.
4. Stop servers.
5. Use ddb to verify that the container is left in shards.
6. Enable the checker.
7. Set policy to --all-interactive.
8. Start the checker and query the checker until the fault is detected.
9. Repair by selecting the destroy option.
10. Query the checker until the fault is repaired.
11. Disable the checker.
12. Run the ddb command and verify that the container is removed from shard.

Create recovery_test_base.py to store methods used across the CR tests.

Add daos faults command support in daos_utils_base.py and daos_utils.py

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_orphan_container ddb_cmd
Test-repeat: 4
Required-githooks: true

### Before requesting gatekeeper:


### Gatekeeper:
